src/pybind/mgr/ssh/module.py

```diff
@@ -984,7 +984,14 @@ def create_osds(self, drive_group):
 
         return self.get_hosts().then(lambda hosts: self._create_osd(hosts, drive_group))
 
-    def remove_osds(self, name):
+    def remove_osds(self, osd_ids):
+        if len(osd_ids) == 1:
+            return self.remove_osd(osd_ids[0])
+        else:
+            return self.remove_osd(osd_ids[0]).then(
+                lambda r: self.remove_osds(osd_ids[1:]))
```

Comment:
conflicts with 

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/ssh/HACKING.rst#i-want-to-call-my-function-for-every-host


```python
@async_map_completion
def remove_osds(self, osd_id):
    return self._remove_daemon(args)

def _search(daemons):
    args = [('osd.%s' % d.service_instance, d.nodename) for d in daemons]
    if not args: raise OrchestratorError('Unable to find osd.%s' % name)
    return args

def remove_osds(self, osd_ids):
    return self._get_services('osd', service_id=name).then(_search).then(_remove_osd)
```